File: gcpc2026/egocentricexpedition/submissions/accepted/Yidi.java
import java.util.Scanner;

public class Yidi {
	public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        for (int t=0; t<T; t++) {
            System.out.println("? 0 1");
            System.out.flush();
            double d1 = sc.nextDouble();

            System.out.println("? 1 1");
            System.out.flush();
            double d2 = sc.nextDouble() * Math.sqrt(0.5);

            double d = Math.abs(d1*d2 / Math.sqrt(d2*d2 + (d1-d2)*(d1-d2)));
            System.out.print("! ");
            System.out.println(Math.round(d*d*4));
            System.out.flush();
        }
	}
}

File: gcpc2026/egocentricexpedition/submissions/accepted/yidi.py
T = int(input())
for t in range(T):
    print("? 0 1", flush=True)
    d1 = float(input())
    print("? 1 1", flush=True)
    d2 = float(input()) * 0.5**0.5

    d = abs(d1*d2 / (d2**2 + (d1-d2)**2)**0.5)
    A = d*d*4
    assert(abs(round(A) - A) < 1e-7)
    print("!", int(round(A)), flush=True)
